src/Components/Debugger.fs

```diff
@@ -360,24 +360,22 @@ module Debugger =
                                     match defaultConfigForProject (p, buildTaskForProject) with
                                     | Some p -> yield p
                                     | None -> ()
-                                })
+                                ])
 
                         return ResizeArray tasks
                     }
 
-                generate ()
-                |> Promise.map Some
-                |> Promise.toThenable
-                |> U2.Case2
-                |> ProviderResult.Some
+                generate () // this bix/unbox is a hack because JS types
+                |> box
+                |> unbox
 
             override x.resolveDebugConfiguration
                 (
                     folder: option<WorkspaceFolder>,
                     debugConfiguration: DebugConfiguration,
                     token: option<CancellationToken>
                 ) =
-                logger.Info $"Evaluating launch settings configurations for workspace2 '{folder}'"
+                logger.Info("Resolving launch settings configuration %O for %O", debugConfiguration, folder)
```

Comment:
One interesting thing here - VSCode calls this method with a null debugConfiguration if you press `f5` and no launch configuration has been configured for this workspace before - maybe we should find the current open editor (if any) or selected project in the tree (if any?) and force that task?